packages/workshop-app/app/utils/diff.server.ts

```diff
@@ -223,38 +263,33 @@ export async function getDiffFiles(app1: App, app2: App) {
 
 	const { stdout: diffOutput } = await execa(
 		'git',
-		['diff', '--no-index', '--name-status', app1CopyPath, app2CopyPath],
+		['diff', '--no-index', '--ignore-blank-lines', app1CopyPath, app2CopyPath],
 		{ cwd: diffTmpDir },
 		// --no-index implies --exit-code, so we need to ignore the error
 	).catch(e => e)
 
 	void fsExtra.remove(app1CopyPath)
 	void fsExtra.remove(app2CopyPath)
 
-	const diffFiles = diffOutput
-		.split('\n')
-		.map(line => {
-			const [status, filePath] = line
-				.split(/\s/)
-				.map(s => s.trim())
-				.filter(typedBoolean)
-			if (!status || !filePath) return null
-			return {
-				status: (status.startsWith('R')
-					? 'renamed'
-					: status === 'M'
-					? 'modified'
-					: status === 'D'
-					? 'deleted'
-					: status === 'A'
-					? 'added'
-					: 'unknown') as 'renamed' | 'moved' | 'deleted' | 'added' | 'unknown',
-				// always use forward separator for our README files
-				path: diffPathToRelative(filePath).replace(/\\/g, '/'),
-			}
-		})
+	const typesMap = {
+		ChangedFile: 'modified',
+		AddedFile: 'added',
+		DeletedFile: 'deleted',
+		RenamedFile: 'renamed',
+	}
+
+	const parsed = parseGitDiff(diffOutput)
+
+	return parsed.files
+		.map(file => ({
+			//  prettier-ignore
+			status: (typesMap[file.type]??'unknown') as 'renamed'|'modified'|'deleted'|'added'|'unknown',
+			// always use forward separator for our README files
+			path: diffPathToRelative(
+				file.type === 'RenamedFile' ? file.pathBefore : file.path,
+			).replace(/\\/g, '/'),
+		}))
 		.filter(typedBoolean)
-	return diffFiles
```

Comment:
Can you explain why this was changed.

Reply:
Accordion to my testing when using git diff with `--name-status` the flag `--ignore-blank-lines` was ignored.
Instead i use `['diff', '--no-index', '--ignore-blank-lines', app1CopyPath, app2CopyPath]` with `parseGitDiff`
and filter the result to include only status and path

Reply:
Makes sense 👍